### DISCUSSION TITLE
MitosisJSX vs Sveltosis ?

### DISCUSSION THREAD

**Discussion by tomByrer** [BuilderIO/jsx-lite#1023] [Ideas]
Seems there has been stealth-addition to Mitosis that allows Sveltosis (Svelte syntax) as input, at least in the demo.
Fortunately there are many examples.
Unfortunately there is not 1:1 of the same examples in JSX, nor is there any documentation.

Could someone help improve onboarding improvements, eg document Sveltosis a bit better (at least Sveltosis vs Svelte & JSX) please?

**Comment by AsyncBanana** [comment]
Please do this; I am trying to decide between MitosisJSX and Sveltosis, but I do not know what compatibility issues Sveltosis might have compared to standard Svelte.

**Reply by tomByrer** [reply]
Further investigation:
Seems [Sveltosis was an external project](https://github.com/sveltosis/sveltosis) that was merged in ~3 months ago.
old [Compatibility chart](https://github.com/sveltosis/sveltosis/tree/main/packages/parser#sveltosisparser) & [Nice demos](https://party.sveltosis.dev/)
I do not see any new Sveltosis features or docs added since.

**Comment by samijaber** [comment]
Hey @tomByrer: you're right, we haven't added any docs about Sveltosis yet, besides the examples in https://mitosois.builder.io/ under the Sveltosis tab. I know Mitosis' docs are way behind its functionality, and we're trying to play catch up 😅

In general, the JSX syntax has much more feature coverage, as the Sveltosis syntax is still new.

Your best bet is to use the fiddle to try different things and see what works.

**Reply by samijaber** [reply]
We would love contributions to add docs around Sveltosis 🙏🏽

**Reply by tomByrer** [reply]
I tried to locally run this repo on one of my laptops, but got too many errors to deal with (Win10, VSCode).
I'll try on another machine, or use Playwright to turn your fiddle into an API ;)

**Reply by samijaber** [reply]
@tomByrer I don't have a Windows machine, but if you can share some of these errors in Discord or Github, we can try to help you. 🙏🏽

**Comment by effulgentsia** [comment]
The Sveltosis tab in https://mitosois.builder.io/ seems broken (none of the templates are generating output). I opened https://github.com/BuilderIO/mitosis/issues/1327 for it.

**Comment by martin-braun** [comment]
This just struck me and it's awesome. I am following BuilderIO a little and liked the idea of a JSX superset, but my personal turn-off is React (I'm not a fan of hooks). Svelte on the other hand has such a better way to solve things (in my opinion), but never had the ecosystem. 

The reason it "strucks" me is, I was giving up on Expo and ReactNative for having React as hard dependency. I wonder what would happen if you throw misosis into the mix. Could it really be possible to build ReactNative components with Svelte? This sounds like a huge deal, but also huge compile times.

I would absolutely start to experiment with this, if I had more time available. Maybe some day, maybe someone else does this in the meantime. Please let me know. Have a nice day! :)

**Reply by tomByrer** [reply]
ReactNative still needs React, regardless on how you obtain that React code ;)
Sure maybe the Svelte-Mitosis can create RN code for you, but if you have issues, you may still need to read the RN/React/hooks code.  Which may be more likely with the more mobile features you use (eg camera).

Ionic/Capacitor is thing to build mobile apps...

**Reply by martin-braun** [reply]
@tomByrer Thanks and yes, Capacitor saved us a lot of times in the past. I had quite a few "can you turn this website into an app?" requests in the past, but this is different. I'm looking for serious ground-up stuff, I would never pick Capacitor, because it just doesn't feel snappy like a native app. Right now I enjoy using Flutter, because it fakes the native part so good that you really can't tell the difference, but it has its own downsides. Thanks for pointing out the obvious challenge of debugging the output of Misosis, that's obvious yet easy to miss. A crucial detail that makes me rather oppose this idea.

**Comment by effulgentsia** [comment]
My comment from Jan 9 has been fixed and https://mitosis.builder.io/playground is working well now, so you can use that to quickly check if the Svelte code that you have in mind gets successfully converted to ReactNative code. For very simple Svelte code, this might work. Mitosis/Sveltosis still seems fairly early stage and unlikely to be able to handle more complex Svelte code (e.g., the kind that you would need hooks for in React).
